windows/RNSVG/CircleView.cpp

```diff
@@ -29,13 +28,25 @@ void CircleView::UpdateProperties(IJSValueReader const &reader, bool forceUpdate
   __super::UpdateProperties(reader, forceUpdate, invalidate);
 }
 
-void CircleView::CreateGeometry(UI::Xaml::CanvasControl const &canvas) {
-  auto const &resourceCreator{canvas.try_as<ICanvasResourceCreator>()};
+void CircleView::CreateGeometry() {
+  auto const root{SvgRoot()};
+  
+  float cx{Utils::GetAbsoluteLength(m_cx, root.ActualWidth())};
+  float cy{Utils::GetAbsoluteLength(m_cy, root.ActualHeight())};
+  float r{Utils::GetAbsoluteLength(m_r, Utils::GetCanvasDiagonal(root.ActualSize()))};
 
-  float cx{Utils::GetAbsoluteLength(m_cx, canvas.Size().Width)};
-  float cy{Utils::GetAbsoluteLength(m_cy, canvas.Size().Height)};
-  float r{Utils::GetAbsoluteLength(m_r, Utils::GetCanvasDiagonal(canvas.Size()))};
+  com_ptr<ID2D1DeviceContext1> deviceContext;
+  copy_to_abi(root.DeviceContext(), *deviceContext.put_void());
 
-  Geometry(Geometry::CanvasGeometry::CreateCircle(resourceCreator, cx, cy, r));
+  com_ptr<ID2D1Factory> factory;
+  deviceContext->GetFactory(factory.put());
+
+  com_ptr<ID2D1EllipseGeometry> geometry;
+  check_hresult(factory->CreateEllipseGeometry(D2D1::Ellipse({cx, cy}, r, r), geometry.put()));
+
+  IInspectable asInspectable;
+  copy_from_abi(asInspectable, geometry.get());
```

Comment:
it's because these types are not WinRT types, so they won't implement IInspectable.
copy_from_abi is just a reinterpret cast, so there's no safety. This operation is in fact undefined behavior :) so instead recommend making m_geometry something other than IInspectable. Either the right base [COM] interface, or IUnknown.